lib/jit/syntax/expression/preset.cpp

```diff
@@ -101,17 +113,27 @@ gemm::args gemm::check(expression_tree::data_type const & tree, size_t root)
             else if(rleft.type==DENSE_ARRAY_TYPE)
             {
                 result.C = &rleft;
-                result.beta = value_scalar(1, dtype);
-                result.alpha = value_scalar(is_add?1:-1,  dtype);
+                if(dtype==isaac::HALF_TYPE){
+                  result.beta = value_scalar(1, isaac::FLOAT_TYPE);
+                  result.alpha = value_scalar(is_add?1:-1,  isaac::FLOAT_TYPE);
+                }else{
+                  result.beta = value_scalar(1, dtype);
+                  result.alpha = value_scalar(is_add?1:-1,  dtype);
+                }
             }
 
             if(rright.type==COMPOSITE_OPERATOR_TYPE)
                 handle_node(tree, right.binary_operator.rhs, result);
             else if(rright.type==DENSE_ARRAY_TYPE)
             {
                 result.C = &rright;
-                result.alpha = value_scalar(1, dtype);
-                result.beta = value_scalar(is_add?1:-1, dtype);
+                if(dtype==isaac::HALF_TYPE){
+                  result.beta = value_scalar(1, isaac::FLOAT_TYPE);
+                  result.alpha = value_scalar(is_add?1:-1,  isaac::FLOAT_TYPE);
+                }else{
+                  result.alpha = value_scalar(1, dtype);
+                  result.beta = value_scalar(is_add?1:-1, dtype);
+                }
             }
         }
         else{
```

Comment:
Again, I would prefer a single declaration of:
numeric_type abtype = (dtype==HALF_TYPE)?:FLOAT_TYPE:dtype;
over all these branches :)